nix/postgres.nix

```diff
@@ -77,32 +107,58 @@ in
           '';
         };
 
-        settings = lib.mkOption {
-          type = with lib.types; attrsOf (oneOf [ bool float int str ]);
-          default = { };
-          description = ''
-            PostgreSQL configuration. Refer to
-            <https://www.postgresql.org/docs/11/config-setting.html#CONFIG-SETTING-CONFIGURATION-FILE>
-            for an overview of `postgresql.conf`.
+        settings =
+          let
+            # Default pg_hba.conf entries
+            defaultHbaConf = [
+              { type = "local"; database = "all"; user = "all"; address = ""; method = "trust"; }
+              { type = "host"; database = "all"; user = "all"; address = "127.0.0.1/32"; method = "trust"; }
+              { type = "host"; database = "all"; user = "all"; address = "::1/128"; method = "trust"; }
+              { type = "local"; database = "replication"; user = "all"; address = ""; method = "trust"; }
+              { type = "host"; database = "replication"; user = "all"; address = "127.0.0.1/32"; method = "trust"; }
+              { type = "host"; database = "replication"; user = "all"; address = "::1/128"; method = "trust"; }
+            ];
 
-            String values will automatically be enclosed in single quotes. Single quotes will be
-            escaped with two single quotes as described by the upstream documentation linked above.
-          '';
-          default = {
-            listen_addresses = config.listen_addresses;
-            port = config.port;
-            unix_socket_directories = lib.mkDefault config.dataDir;
+            # Merge the default pg_hba.conf entries with the user-defined entries
+            hbaConf = defaultHbaConf ++ config.hbaConf;
+
+            # Convert the pgHbaConf array to a string
+            hbaConfString = ''
+              # Generated by Nix
+              ${"# TYPE\tDATABASE\tUSER\tADDRESS\tMETHOD\n"}
+              ${lib.concatMapStrings (cnf: "  ${cnf.type}\t${cnf.database}\t${cnf.user}\t${cnf.address}\t${cnf.method}\n") hbaConf}
+            '';
+            hbaDrv = pkgs.runCommand "pg_hba.conf" { } ''
+              echo "${hbaConfString}" | ${pkgs.util-linux}/bin/column -t -s $'\t' > $out
+            '';
```

Comment:
I'd put this entire thing an `internal` (and `readOnly`) option alongside `config.hbaConf`. Maybe call it `config.hbaConfFile`. Doing so decouples the hba stuff from the `settings` option thus making it simpler to read.